### ISSUE TITLE
Digest fails to build when using `std` feature

### ISSUE THREAD

**Issue by Wallacoloo** [RustCrypto/traits#1]
When building with the digest crate with the `std` feature enabled, I get the following errors:

```rust
traits/digest/src/$ cargo build --features std
   Compiling digest v0.6.0 (file:///home/colin/src/traits/digest)
error[E0576]: cannot find associated type `OutputSize` in trait `Digest`
  --> digest.rs:64:39
   |
64 |         -> io::Result<Output<<Self as Digest>::OutputSize>>
   |                                       ^^^^^^^^^^^^^^^^^^^ not found in `Digest`

error[E0433]: failed to resolve. Use of undeclared type or module `core`
  --> lib.rs:49:22
   |
49 | pub trait XofReader: core::marker::Sized {
   |                      ^^^^^^^^^^^^^^^^^^^ Use of undeclared type or module `core`
```

Building without std works fine. I notice that changing the first few lines in Digest's `lib.rs` fixes compilation for `cargo build --features std`:

```rust
#![no_std]
extern crate generic_array;
```

Instead of
```rust
#![cfg_attr(not(feature = "std"), no_std)]
extern crate generic_array;
```

But I'm not sure what the full implications of always passing the `no_std` option to `generic_array` is.

**Comment by newpavlov** [comment]
It was a bug which I missed due to the insufficient testing... It should be fixed now. I've published new version `0.6.1` with the fix and yanked `0.6.0`.

**Comment by Wallacoloo** [comment]
@newpavlov Thanks for the quick fix! Confirmed working on my end.
